Sources/LiveKit/Support/FloatRingBuffer.swift

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2024 LiveKit
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import Foundation
+
+// Simple ring-buffer used for internal audio processing. Not thread-safe.
+class FloatRingBuffer {
```

Comment:
doesn't really matter but you could call this `RingBuffer<T: Numeric>` and replace all of the `Float` with `T` below and it would still work as-is.